### PR TITLE
Add documentation code sample syntax highlighting

### PR DESCRIPTION
**Pull request by cubrr**
Modifies the code samples in the documentation which are indented with four spaces to be wrapped inside three backticks instead. This allows us to specify the language of the code sample for nice syntax highlighting.

##### Before:
```
{
    //check if profiler set flag, pin profiler
    BEGIN_PIN_PROFILER(CORProfilerTrackModuleLoads());

    //call the wrapper around the profiler's callback implementation
    g_profControlBlock.pProfInterface->ModuleLoadStarted((ModuleID) this);          

    //unpin profiler
    END_PIN_PROFILER();
}
```

##### After
```c++
{
    //check if profiler set flag, pin profiler
    BEGIN_PIN_PROFILER(CORProfilerTrackModuleLoads());

    //call the wrapper around the profiler's callback implementation
    g_profControlBlock.pProfInterface->ModuleLoadStarted((ModuleID) this);          

    //unpin profiler
    END_PIN_PROFILER();
}
```

### CODE REVIEW

File: Documentation/building/linux-instructions.md [hunk 1]
```diff
@@ -36,7 +36,9 @@ ellismg@linux:~$ sudo apt-get update
 
 Then install the packages you need:
 
-`ellismg@linux:~$ sudo apt-get install cmake llvm-3.5 clang-3.5 lldb-3.6 lldb-3.6-dev libunwind8 libunwind8-dev gettext`
+```bash
+ellismg@linux:~$ sudo apt-get install cmake llvm-3.5 clang-3.5 lldb-3.6 lldb-3.6-dev libunwind8 libunwind8-dev gettext
+```
```
Comment (jasonwilliams200OK) [inline Documentation/building/linux-instructions.md]: Please remove `ellismg@linux:~$` prefixes to make bash commands copy-paste-able.

Comment (cubrr) [inline Documentation/building/linux-instructions.md]: I would imagine that leaving those in were a conscious choice to show the working directory later in the guide (for example, [line 127 of `linux-instructions.md`](https://github.com/cubrr/coreclr/commit/b246686a85f59d4a3ed578706621dcc2a55e777e#diff-dfd54bb8cec34d34e6f3c9bab15d76acR127)):

```bash
 ellismg@linux:~/coreclr-demo/packages$ curl -L -O https://nuget.org/nuget.exe
 ```

I would prefer to keep this pull request about the syntax highlighting, not changing the examples themselves.

Comment (ellismg) [inline Documentation/building/linux-instructions.md]: Yes.  I wanted to be very careful to show what directory I was in at all times.

**Comment by adityamandaleeka** [comment]
It looks like this PR just fell off the radar last year. @cubrr sorry about that. 

These changes look fine to me. Unfortunately, it looks like there are now merge conflicts that need to be resolved. If you're still interested in doing this, can you please update this with the conflicts resolved? Otherwise, I'll close the PR in a few days (feel free to open another PR if this one is already closed by the time you see this).

**Pull request closed without merging**